packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -32,135 +30,125 @@ import { pushUniqueRoute } from 'utils/route'
 import styles from './PremiumContentPurchaseModal.module.css'
 import { PurchaseContentFormFields } from './components/PurchaseContentFormFields'
 import { PurchaseContentFormFooter } from './components/PurchaseContentFormFooter'
-import { CUSTOM_AMOUNT, AMOUNT_PRESET } from './components/constants'
-import { PayExtraPreset } from './components/types'
-import {
-  PurchaseContentSchema,
-  PurchaseContentValues
-} from './components/validation'
-import { getExtraAmount } from './hooks'
+import { usePurchaseContentFormState } from './hooks/usePurchaseContentFormState'
 
-const { startRecoveryIfNecessary, cleanup } = buyUSDCActions
+const { startRecoveryIfNecessary, cleanup: cleanupUSDCRecovery } =
+  buyUSDCActions
 
 const messages = {
   completePurchase: 'Complete Purchase'
 }
-const { startPurchaseContentFlow } = purchaseContentActions
-const { getPurchaseContentFlowStage, getPurchaseContentError } =
-  purchaseContentSelectors
 
-export const PremiumContentPurchaseModal = () => {
+const RenderForm = ({
+  onClose,
+  track
+}: {
+  onClose: () => void
+  track: PurchasableTrackMetadata
+}) => {
   const dispatch = useDispatch()
-  const stage = useSelector(getPurchaseContentFlowStage)
-  const error = useSelector(getPurchaseContentError)
-  const isUnlocking = !error && isContentPurchaseInProgress(stage)
+  const {
+    permalink,
+    premium_conditions: {
+      usdc_purchase: { price }
+    }
+  } = track
+  const state = usePurchaseContentFormState({ price })
+  const { error, isUnlocking, purchaseSummaryValues, stage } = state
 
-  const initialValues: PurchaseContentValues = {
-    [CUSTOM_AMOUNT]: undefined,
-    [AMOUNT_PRESET]: PayExtraPreset.NONE
-  }
+  // Attempt recovery once on re-mount of the form
+  useEffect(() => {
+    dispatch(startRecoveryIfNecessary)
+  }, [dispatch])
+
+  const handleClose = useCallback(() => {
+    dispatch(cleanupUSDCRecovery())
+    onClose()
+  }, [dispatch, onClose])
```

Comment:
@sddioulde I had to split up the logic for recovery and do part of it here and part of it in the place where we actually read the balance for the form.

When we initially created the useUSDCBalance hook, the idea was that we didn't want to use a slice to store the info and sync it across all readers, since the logic to do the for $AUDIO got quite complicated. But that now means every instance of `useUSDCBalance` has its own local cached value that is only refreshed if you refresh that individual hook.
It also means that the `cleanup` function could be called from one component and reset the recovery process happening in another.

So, I think at the moment this isn't likely to exhibit any bad behavior, but it could very easily do so in the future. We will probably need to make recovery its own slice and maybe move the fetching into audius-query so that all readers subscribe to the same value.